pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -290,6 +298,14 @@ def call_configure_sharded_model_hook(self) -> bool:
     def call_configure_sharded_model_hook(self, mode: bool) -> None:
         self._call_configure_sharded_model_hook = mode
 
+    @abstractmethod
+    def teardown(self) -> None:
+        """
+        This method is called to teardown the training process.
+        It is the right place to release memory and free other ressources.
```

Comment:
```suggestion
        It is the right place to release memory and free other resources.
```

Reply:
> Any way we can test the teardowns?

for gpu, might be tricky, maybe could check the memory at the end.